pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -250,7 +251,8 @@ func (o *operatingSystemConfig) reconcile(ctx context.Context, reconcileFn func(
 }
 
 type poolHash struct {
-	Pools []poolHashEntry
+	Pools    []poolHashEntry
+	Migrated *bool `yaml:"migrated,omitempty"`
```

Comment:
Should we add a TODO statement to drop this field after `v1.99` got released?

Reply:
That depends a bit on the assumption for how long the OSC of a cluster might have failed to reconcile. The field should only be removed once the reconcile has happened once. What is the usual approach in that regard? Add a remark to the changelog that all clusters must be reconciled once before upgrading?

Reply:
> What is the usual approach in that regard? Add a remark to the changelog that all clusters must be reconciled once before upgrading?

Yes, either this, or we implement code that reconciles OSCs early in the flow for existing shoots. We can also just remove it when we drop support for the currently lowest Kubernetes version, i.e., 1.25?

Reply:
I've added a flow task to trigger the initial pool-hashes secret migration as one of the first steps while reconciling a shoot. PTAL.